components/scream/src/physics/shoc/shoc_inputs_initializer.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef SCREAM_SHOC_INPUTS_INITIALIZER_HPP
+#define SCREAM_SHOC_INPUTS_INITIALIZER_HPP
+
+#include "share/field/field_initializer.hpp"
+#include "physics/share/physics_functions.hpp"
+#include "physics/shoc/shoc_constants.hpp"
+#include "physics/shoc/shoc_functions.hpp"
+
+namespace scream {
+
+class SHOCInputsInitializer : public FieldInitializer
+{
+  using SHF            = shoc::Functions<Real, DefaultDevice>;
+  using Spack          = typename SHF::Spack;
+  using view_1d_scalar = typename SHF::view_1d<Real>;
+  using view_2d        = typename SHF::view_2d<Spack>;
+
+public:
+
+  virtual ~SHOCInputsInitializer () = default;
+
+  // The name of the initializer
+  std::string name () const { return "SHOCInputsInitializer"; }
+
+  // Initialize fields
+  void initialize_fields ();
```

Comment:
You can add the `override` keyword here to make clear this is an overridden method.